drivers/usb/device/Kconfig

```diff
@@ -197,6 +197,41 @@ config USB_DC_IT82XX2
 	help
 	  ITE IT82XX2 USB Device Controller Driver
 
+config USB_DC_NUMAKER
+	bool "Nuvoton NuMaker USB 1.1 device controller"
+	default y
+	select HAS_NUMAKER_USBD
+	depends on DT_HAS_NUVOTON_NUMAKER_USBD_ENABLED
+	help
+	  Enable Nuvoton NuMaker USB 1.1 device controller driver
+
+if USB_DC_NUMAKER
+
+config USB_DC_NUMAKER_ALLOW_ISO_IN_OUT_SAME_NUM
+	bool "Allow Isochronous IN/OUT endpoints to be of same numbers"
+	default y
+	depends on $(dt_nodelabel_bool_prop,usbd,disallow-iso-in-out-same-number)
+	help
+	  Enable allowing Isochronous IN/OUT endpoints to be assigned
+	  the same endpoint numbers. Note this can only work for not more than
+	  one Isochronous IN endpoint enabled, or behavior is undefined.
```

Comment:
Based on this description it is not so clear to me what is the real issue there, i.e. what are the actual hardware limitations.

Why cannot the endpoint assignemnt function simply fail if it cannot assign the endpoints? When would you want to have this KConfig option enabled and what does it really give?

Reply:
refined the description, some soc series doesn't allow ISO IN/OUT endpoints to be assigned the same endpoint numbers, for example, 0x82/0x02 for ISO IN/OUT. On this soc series with this limitation, this Kconfig option defaults to enabled so that [audio samples](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/subsys/usb/audio) can run without change because they assign ISO IN/OUT endpoints to 0x81/0x01.

Reply:
yes, zephyr usb device stack allows endpoint address  to adjust based on `usb_dc_ep_check_cap()`. removed this Kconfg `USB_DC_NUMAKER_ALLOW_ISO_IN_OUT_SAME_NUM` and related code and re-implemented `usb_dc_ep_check_cap()` to address this h/w limit.